topostats/grains.py

```diff
@@ -1577,3 +2032,268 @@ def merge_classes(
+                square_flat_bounding_box[1] : square_flat_bounding_box[3],
+                :,
+            ]
+
+            # Update background class to reflect the removal of any non-connected grains
+            grain_cropped_tensor = Grains.update_background_class(grain_mask_tensor=grain_cropped_tensor)
+
+            assert grain_cropped_image.shape[0] == grain_cropped_image.shape[1]
+            assert grain_cropped_tensor.shape[0] == grain_cropped_tensor.shape[1]
+            # Check that the bounding box is square
+            bounding_box_shape = (
+                square_flat_bounding_box[2] - square_flat_bounding_box[0],
+                square_flat_bounding_box[3] - square_flat_bounding_box[1],
+            )
+            assert bounding_box_shape[0] == bounding_box_shape[1]
+            # Check bounding box shape is same as image shape and first two dimensions of tensor
+            assert bounding_box_shape == grain_cropped_image.shape
+            assert bounding_box_shape == (grain_cropped_tensor.shape[0], grain_cropped_tensor.shape[1])
+
+            graincrops[grain_number] = GrainCrop(
+                image=grain_cropped_image,
+                mask=grain_cropped_tensor,
+                padding=padding,
+                bbox=square_flat_bounding_box,
+                pixel_to_nm_scaling=pixel_to_nm_scaling,
+                filename=filename,
+            )
+
+        return graincrops
+
+    @staticmethod
+    def graincrops_remove_objects_too_small_to_process(
+        graincrops: dict[int, GrainCrop],
+        min_object_size: int,
+        min_object_bbox_size: int,
+    ) -> dict[int, GrainCrop]:
+        """
+        Remove objects that are too small to process from each class of the grain crops.
+
+        Parameters
+        ----------
+        graincrops : dict[int, GrainCrop]
+            Dictionary of grain crops.
+        min_object_size : int
+            Minimum object size to keep (pixels).
+        min_object_bbox_size : int
+            Minimum object bounding box size to keep (pixels^2).
+
+        Returns
+        -------
+        dict[int, GrainCrop]
+            Dictionary of grain crops with objects too small to process removed.
+        """
+        for _grain_number, graincrop in graincrops.items():
+            # Iterate over the classes
+            for class_index in range(1, graincrop.mask.shape[2]):
+                # Get the binary mask for the class
+                class_mask = graincrop.mask[:, :, class_index]
+
+                # Label the regions
+                labelled_regions = Grains.label_regions(class_mask)
+                region_properties = Grains.get_region_properties(labelled_regions)
+
+                # Iterate over the regions
+                for region in region_properties:
+                    # Get the region mask
+                    region_mask = labelled_regions == region.label
+                    region_bbox = region.bbox
```

Comment:
I think since its just accessing elements of `region.bbox` we could do that directly without losing any readability.

```suggestion
```